src/main/scala/edu/ie3/simona/event/listener/DBFSListener.scala

```diff
@@ -0,0 +1,141 @@
+import breeze.math.Complex
+import edu.ie3.simona.agent.grid.SweepValueStore
+import edu.ie3.simona.event.Event
+import edu.ie3.simona.event.listener.DBFSListener.DBFSResults
+import edu.ie3.simona.logging.SimonaActorLogging
+import edu.ie3.simona.sim.SimonaSim.ServiceInitComplete
+import edu.ie3.simona.util.ResultFileHierarchy
+
+import java.io.{File, PrintWriter}
+import java.util.UUID
+
+object DBFSListener {
+
+  final case class DBFSResults(
+      senderName: String,
+      tick: Long,
+      sweepData: Map[Int, SweepValueStore]
+  ) extends Event
+
+  def props(
+      resultFileHierarchy: ResultFileHierarchy,
+      supervisor: ActorRef
+  ): Props =
+    Props(
+      new DBFSListener(
+        resultFileHierarchy,
+        supervisor
+      )
+    )
+}
+
+final class DBFSListener(
+    resultFileHierarchy: ResultFileHierarchy,
+    supervisor: ActorRef
+) extends SimonaListener
+    with SimonaActorLogging {
+
+  // unblock supervisor to go on with simulation
+  supervisor ! ServiceInitComplete
+
+  //  private val outputFile = resultFileHierarchy.rawOutputDataDir + File.separator + "dbfs_kpi.csv"
+  private val outputWriter =
+    scala.collection.mutable.Map.empty[String, PrintWriter]
+  private val outputHeadline =
+    scala.collection.mutable.Map.empty[String, Vector[UUID]]
+  private val sep = ";"
+
+  override def receive: Receive = {
+    case DBFSResults(senderName, tick, sweepData) =>
+      handleSweepData(senderName, tick, sweepData)
+    case _ => // drop other messages
+  }
+
+  private def handleSweepData(
+      senderName: String,
+      tick: Long,
+      sweepData: Map[Int, SweepValueStore]
+  ): Unit = {
+
+    val writer = outputWriter.getOrElse(
+      senderName, {
+        // create writer
+        val outputFile =
+          s"${resultFileHierarchy.rawOutputDataDir + File.separator}dbfs_v_mag_pu_$senderName.csv"
+        val writer = new PrintWriter(outputFile)
+        // create and write headline
+        sweepData.headOption
+          .map(sweepData => createHeadline(senderName, sweepData._2))
+          .foreach(writer.write)
+        writer.flush()
+        // add data to outputWriterMap
+        this.outputWriter += (senderName -> writer)
+        writer
+      }
+    )
+
+    // write row data
+    sweepData.foreach { case (sweepNo, sweepValueStore: SweepValueStore) =>
+      createEntryRow(senderName, sweepNo, tick, sweepValueStore).foreach(
+        writer.write
+      )
+      writer.flush()
+    }
+  }
+
+  def createHeadline(
+      senderName: String,
+      sweepValueStore: SweepValueStore
+  ): String = {
+    // tick, sweepNo, local_pf_counter, {shortNodeUuids}
+    val uuids = sweepValueStore.sweepData
+      .map(_.nodeUuid.toString.substring(0, 8))
+      .mkString(sep)
+    // we want to keep this order for later
+    this.outputHeadline += (senderName -> sweepValueStore.sweepData.map(
+      _.nodeUuid
+    ))
+    s"tick${sep}sweep_no${sep}local_pf_counter${sep}$uuids\n"
```

Comment:
That is generally a not too bad idea. However, in this case, the data scheme is not fixed (cf. variable amount of "last" columns depending on the nodes of the targeted sub grid).

I agree with you, that this is not super nice. Another option would be to alter the csv entries from `tick,sweep_no,local_pf_counter,<node_uuid_0>,...` to `tick,sweep_no,local_pf_counter,node_uuid_0` and then have multiple lines instead of one line. What do you think, @johanneshiry? Would that fit the needs of the use case you wanted to address with this feature?

We shouldn't accept code here, that doesn't fit our quality standards.

Reply:
This is indeed an add-hoc solution and I would prefer not to have it in the code in that way. I would also not like to use DataSink from PSDM as the purpose of this interface is quite different to the ones we need here.

I may provide a draft for a new, refactored structure if I have time. This structure could include 
- akka typed actor 
- trait to write case classes to csv out of the box (cf. csv-support trait) 

What do you think @ckittl @sebastian-peter

Reply:
Additionally, this could resolve the mutable agent state...

Reply:
> This is indeed an add-hoc solution and I would prefer not to have it in the code in that way. I would also not like to use DataSink from PSDM as the purpose of this interface is quite different to the ones we need here.
> 
> I may provide a draft for a new, refactored structure if I have time. This structure could include
> 
> * akka typed actor
> * trait to write case classes to csv out of the box (cf. csv-support trait)
> 
> What do you think @ckittl @sebastian-peter

LGTM